applications/ConstitutiveLawsApplication/custom_constitutive/composites/traction_separation_law.cpp

```diff
@@ -621,14 +610,78 @@ int TractionSeparationLaw3D<TDim>::Check(
     const ProcessInfo& rCurrentProcessInfo
     ) const
 {
-    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(INTERFACIAL_NORMAL_STRENGTH)) << "INTERFACIAL_NORMAL_STRENGTH is not a defined value" << std::endl;
-    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(INTERFACIAL_SHEAR_STRENGTH)) << "INTERFACIAL_SHEAR_STRENGTH is not a defined value" << std::endl;
-    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(MODE_ONE_FRACTURE_ENERGY)) << "MODE_ONE_FRACTURE_ENERGY is not a defined value" << std::endl;
-    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(MODE_TWO_FRACTURE_ENERGY)) << "MODE_TWO_FRACTURE_ENERGY is not a defined value" << std::endl;
-    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(TENSILE_INTERFACE_MODULUS)) << "TENSILE_INTERFACE_MODULUS is not a defined value" << std::endl;
-    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(SHEAR_INTERFACE_MODULUS)) << "SHEAR_INTERFACE_MODULUS is not a defined value" << std::endl;
-    KRATOS_ERROR_IF(rMaterialProperties[MODE_ONE_FRACTURE_ENERGY] < 0.0) << "MODE_ONE_FRACTURE_ENERGY is negative." << std::endl;
-    KRATOS_ERROR_IF(rMaterialProperties[MODE_TWO_FRACTURE_ENERGY] < 0.0) << "MODE_TWO_FRACTURE_ENERGY is negative." << std::endl;
+    // Check if input parameters are completely defined
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(INTERFACIAL_NORMAL_STRENGTH) || rMaterialProperties.Has(INTERFACIAL_NORMAL_STRENGTH_VECTOR)) << "INTERFACIAL_NORMAL_STRENGTH is not a defined value" << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(INTERFACIAL_SHEAR_STRENGTH) || rMaterialProperties.Has(INTERFACIAL_SHEAR_STRENGTH_VECTOR)) << "INTERFACIAL_SHEAR_STRENGTH is not a defined value" << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(MODE_ONE_FRACTURE_ENERGY) || rMaterialProperties.Has(MODE_ONE_FRACTURE_ENERGY_VECTOR)) << "MODE_ONE_FRACTURE_ENERGY is not a defined value" << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(MODE_TWO_FRACTURE_ENERGY) || rMaterialProperties.Has(MODE_TWO_FRACTURE_ENERGY_VECTOR)) << "MODE_TWO_FRACTURE_ENERGY is not a defined value" << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(TENSILE_INTERFACE_MODULUS) || rMaterialProperties.Has(TENSILE_INTERFACE_MODULUS_VECTOR)) << "TENSILE_INTERFACE_MODULUS is not a defined value" << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(SHEAR_INTERFACE_MODULUS) || rMaterialProperties.Has(SHEAR_INTERFACE_MODULUS_VECTOR)) << "SHEAR_INTERFACE_MODULUS is not a defined value" << std::endl;
+
+    // Check the size of the vectors
+    if (rMaterialProperties.Has(INTERFACIAL_NORMAL_STRENGTH_VECTOR)) {
+        KRATOS_ERROR_IF(rMaterialProperties[INTERFACIAL_NORMAL_STRENGTH_VECTOR].size() != (rMaterialProperties[LAYER_EULER_ANGLES].size() / 3.0) - 1.0) << "INTERFACIAL_NORMAL_STRENGTH_VECTOR badly defined" << std::endl;
```

Comment:
hmm careful because we should divide over 3 (`int `division not 3.0) and then substract 1. Otherwise we are doing shady type conversions